tests/models/test_taskinstance.py

```diff
@@ -1163,6 +1163,33 @@ def test_xcom_push_flag(self):
         ti.run()
         assert ti.xcom_pull(task_ids=task_id, key=models.XCOM_RETURN_KEY) is None
 
+    def test_xcom_push_with_a_specified_key_in_operator(self):
+        """
+        Tests the option for Operators to push XComs with a given key
+        """
+        key = 'mykey'
+        value = 'hello'
+        task_id = 'test_no_xcom_push'
+        dag = models.DAG(dag_id='test_xcom')
+
+        # XCom pushed with key=mykey
+        task = PythonOperator(
+            task_id=task_id,
+            dag=dag,
+            python_callable=lambda: value,
+            xcom_key=key,
```

Comment:
I don't think this should a property of BaseOperator -- _if_ we have this feature (and I'm still not sure why we need it) then it should be solely on the TaskFlow decorator, not all operators.

Reply:
I have provided answers to why we may need it in all operators. Check https://github.com/apache/airflow/pull/16823#issuecomment-875037822